pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -489,21 +503,59 @@ func (bfs *Service) IsActive() bool {
 	return bfs.isActive.Load()
 }
 
+// retry function with exponential backoff.
+func (bfs *Service) retry(action func() error) error {
+	var (
+		err     error
+		backoff = initialBackoff
+		timer   = time.NewTimer(0)
+	)
+	defer timer.Stop()
```

Comment:
I'd say that it should be
```go
defer func(){
    if !timer.Stop() {
        <- timer.C
    }
}()
```
Just in case if we'll change the main routine code some day.